.github/workflows/pr-vrt-comment.yml

```diff
@@ -34,134 +31,104 @@ jobs:
       # - see @{link file://./../scripts/prepare-vr-screenshots-for-upload.js#43}
       # - see @{link file://./pr-vrt.yml#56}
-      - uses: actions/download-artifact@v4
+
+      - name: Check for vrscreenshot artifact
+        id: check_artifact
+        uses: actions/github-script@v7
         with:
-          name: vrscreenshot
-          path: ./screenshots
-          run-id: ${{ github.event.workflow_run.id }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.payload.workflow_run.id
+            });
+
+            const artifactExists = artifacts.data.artifacts.some(artifact => artifact.name === "vrscreenshot");
+
+            if (artifactExists) {
+              core.info("Artifact 'vrscreenshot' found.");
+              core.exportVariable("run_diff", "true");
+            } else {
+              core.info("Artifact 'vrscreenshot' not found.");
+              core.exportVariable("run_diff", "false");
+            }
 
       - uses: actions/download-artifact@v4
+        if: ${{ env.run_diff == 'true' }}
         with:
-          name: pr-number
-          path: ./results
+          name: vrscreenshot
+          path: ./screenshots
           run-id: ${{ github.event.workflow_run.id }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Load PR number
+      - name: Save metadata
         uses: actions/github-script@v7
-        id: pr_number
+        if: ${{ env.run_diff == 'true' }}
         with:
           script: |
-            const run = require('./.github/scripts/validate-pr-number');
-            const result = run({filePath:'results/pr.txt'});
-            return result;
-          result-encoding: string
+            const fs = require('fs');
 
-      - name: VR App - Create Policy
-        run: |
-          echo "MAKE THIS STEP WORK"
+            const data = {
+              GITHUB_RUN_ID: context.payload.workflow_run.id,
+              GITHUB_REF: `refs/pull/${context.payload.workflow_run.pull_requests[0].number}/merge`,
+
+              GITHUB_SHA: context.payload.workflow_run.pull_requests[0].head.sha,
+              GITHUB_HEAD_REF: context.payload.workflow_run.pull_requests[0].head.ref,
+
+              GITHUB_REPOSITORY: context.payload.workflow_run.repository.full_name,
+
+              GITHUB_BASE_REF: context.payload.workflow_run.pull_requests[0].base.ref,
+              GITHUB_WORKFLOW: context.payload.workflow_run.name
+            };
+
+            console.log("Metadata", data);
+
+            fs.writeFileSync('metadata.json', JSON.stringify(data, null, 2));
 
       - name: Login via Azure CLI
-        uses: azure/login@a65d910e8af852a8061c627c456678983e180302 # v2.2.0
+        uses: azure/login@a65d910e8af852a8061c627c456678983e180302
```

Comment:
please use patterns that mention version of this sha https://github.com/microsoft/fluentui/blob/master/.github/workflows/pr-website-deploy-comment.yml#L62

Reply:
Updated